### ISSUE TITLE
A small bug of KaTeX plugin

### ISSUE THREAD

**Issue by yuzijie** [Jermolene/TiddlyWiki5#2500]
Hi, I have found a small bug of KaTeX plugin which is the vector symbol cannot be displayed.

From KaTeX official website
![katex](https://cloud.githubusercontent.com/assets/2365206/16913661/6038505a-4cec-11e6-9275-cddc58d64b01.png)

From Tiddlywiki KaTeX demo page
![tiddlywiki](https://cloud.githubusercontent.com/assets/2365206/16913664/645990f4-4cec-11e6-9883-8af4c84f5da5.png)

**Comment by Jermolene** [comment]
Thanks @yuzijie I'm investigating. One issue is that the demo on the KaTeX site actually wraps the text you enter into the string `\displaystyle{<typedformula>}`. After allowing for that I'm finding that TW and the KaTeX demo are generating exactly the same HTML, so I think it might be a fonts issue. Will continue investigating.

**Comment by Jermolene** [comment]
Hi @yuzijie what browser are you using? After a bit more investigation, I think this might be a Chrome bug; in my tests, TiddlyWiki renders correctly in Firefox:

<img width="608" alt="screenshot 2016-07-19 16 04 02" src="https://cloud.githubusercontent.com/assets/174761/16954868/6ffdb79e-4dca-11e6-858a-8f77f28df21e.png">

It's not clear why the KaTeX demo site isn't encountering the same bug, I'll continue investigations.

**Comment by yuzijie** [comment]
Hi @Jermolene, It is indeed a Webkit bug! I never use Firefox, so I assumed that all browsers should have the same problem. However, I did some investigation today, and found that this bug is related to  webkit browsers only, except for Safari, which doesn’t apply the correct length of the arrow. Then accidentally, I found this weird thing:

![bug](https://cloud.githubusercontent.com/assets/2365206/17006647/9c33b2f0-4ee2-11e6-8b22-d6a6ff10f481.png)

It proves my point.

**Comment by yuzijie** [comment]
I think the best solution for this is to add `text-rendering: auto;` in `.katex` class.

**Comment by Jermolene** [comment]
Thank you @yuzijie for your brilliant detective work, the fix works for me.

**Comment by flarr** [comment]
Hi,


I am having this problem with my TiddlyWiki. Vector arrows do not display, whether I open it in Chrome, Edge, Internet Explorer or Firefox.

I am using the latest version, 5.1.17, of both TiddlyWiki and the KaTeX plugin.

I tried to switch from "text-rendering: optimizeLegibility;" to "text-rendering: auto;" within all the KaTeX parts of the .html file, but it didn't help. @yuzijie : What exact part of the .html file is the .katex class you mention, in which I should add "text-rendering: auto;"?

Is nobody else still experiencing this?


Sincerely,

**Comment by bluehood** [comment]
Hi, still experiencing this (I'm on firefox).

I created a new tiddler `KatexStylesheet` with tag `$:/tags/Stylesheet` and content
```
.katex {
    text-rendering: auto;
}
```
and, although it seems that the CSS is being picked up (checked e.g. by changing katex background to red), I still cannot see any arrows over `\vec{x}`.

**Comment by pmario** [comment]
@bluehood ... The problem seems to be from SVG max-width definition from TW. 

```
.katex svg {
  max-width: initial;
}
```

This does the trick for me in FF. ..  The `text-rendering: auto;` imo isn't needed, since it is applied already that way.

**Comment by pmario** [comment]
I was testing with: https://tiddlywiki.com/plugins/tiddlywiki/katex/

So if you did include other plugins, which include CSS settings it may be different results!

**Comment by pmario** [comment]
@Jermolene ... IMO this bug should be reopened an fixed.

**Comment by bluehood** [comment]
@pmario thank you so much, `max-width: initial` did the trick!

**Comment by Jermolene** [comment]
Thanks @bluehood @pmario I've pushed a fix for v5.1.20 -- 94d18a2f1
